dimod/constrained.py

```diff
@@ -852,6 +856,136 @@ def vartype(self, v: Variable) -> Vartype:
         """The vartype of the given variable."""
         return self.objective.vartype(v)
 
+    @classmethod
+    def from_lp_file(cls,
+                     fp: Union[BinaryIO, ByteString],
+                     lower_bound_default: Optional[int] = None,
+                     upper_bound_default: Optional[int] = None) -> "ConstrainedQuadraticModel":
+        """Create a CQM model from a LP file.
+
+        Args:
+            fp: file-like object in the LP format
+            lower_bound_default: in case lower bounds for integer are not given, this will be the default
+            upper_bound_default: in case upper bounds for integer are not given, this will be the default
+
+        Returns:
+            The model encoded in the LP file, converted to ConstrainedQuadraticModel dimod object
+        """
+        grammar = make_lp_grammar()
+        parse_output = grammar.parseFile(fp)
+        obj = get_variables_from_parsed_lp(parse_output, lower_bound_default, upper_bound_default)
+
+        cqm = ConstrainedQuadraticModel()
+
+        # parse and set the objective
+        obj_coefficient = 1
+        for oe in parse_output.objective:
+
+            if isinstance(oe, str):
+                if oe in obj_senses.keys():
+                    obj_coefficient = obj_senses[oe]
```

Comment:
where is `obj_senses` defined? I may be just missing it but I think this code block is broken (the unittests should probably explore this if condition).

Reply:
BTW, you can see what your unittests cover by running `coverage run -m unittest` followed by `coverage html`. Good coverage is necessary but not sufficient for good unittests :smile: